src/Support/SupportData.php

```diff
@@ -358,20 +327,25 @@ public function get_error_log() {
 			];
 		}
 
-		$file        = file( $error_log_path );
-		$max_lines   = max( 0, count( $file ) - 200 );
-		$file_length = count( $file );
-		$contents    = [];
+		$no_of_lines = 200;
+
+		$file  = fopen( $error_log_path, 'r' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_fopen
+		$lines = [];
 
-		for ( $i = $max_lines; $i < $file_length; $i ++ ) {
-			if ( ! empty( $file[ $i ] ) ) {
-				$contents[] = sanitize_text_field( $file[ $i ] );
+		while ( ! feof( $file ) ) {
+			$line = fgets( $file );
+			array_push( $lines, $line );
+			$line_count = count( $lines );
+			if ( $line_count > $no_of_lines ) {
+				array_shift( $lines );
 			}
 		}
 
+		fclose( $file );
```

Comment:
It'd be good to check if the file is a resource that can be read from before attempting to read from it:

```suggestion
		if ( is_resource( $file ) ) {
			while ( ! feof( $file ) ) {
				$line = fgets( $file );
				array_push( $lines, $line );
				$line_count = count( $lines );
				if ( $line_count > $no_of_lines ) {
					array_shift( $lines );
				}
			}

			fclose( $file );
		}
```

Reply:
Addressed in [39e51e6](https://github.com/ampproject/amp-wp/pull/6627/commits/39e51e68a80dc30f28b041c21f51ab49886ec180)